### PR TITLE
﻿Update autorest version to 3.0.0-beta.20210928.1

### PR DESCRIPTION
**Pull request by AlexanderSher**

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Experimental/api/Azure.Messaging.EventHubs.Experimental.netstandard2.0.cs
Comment (jsquire) [inline sdk/eventhub/Azure.Messaging.EventHubs.Experimental/api/Azure.Messaging.EventHubs.Experimental.netstandard2.0.cs:1]: This isn't generated and shouldn't be part of this change set.   Please revert.

**Review by jsquire — CHANGES_REQUESTED** [review]
Looks like some unrelated bits of Event Hubs got caught up in here.   Can we please revert those?

File: sdk/search/Azure.Search.Documents/src/autorest.md
Comment (annelo-msft) [inline sdk/search/Azure.Search.Documents/src/autorest.md:19]: Why do we need to add a flag to HLCs for an LLC change?  It feels like these two things should be independent of each other.

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/api/Azure.Security.ConfidentialLedger.netstandard2.0.cs
Comment (annelo-msft) [inline sdk/confidentialledger/Azure.Security.ConfidentialLedger/api/Azure.Security.ConfidentialLedger.netstandard2.0.cs:20]: How are we handling breaking changes we make to LLC libraries?

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs
Comment (annelo-msft) [inline sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs:61]: Is there a plan for, or issue tracking the work for, bringing these tests back?

Comment (christothes) [inline sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs:61]: why?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/IContainerRegistryAuthenticationClient.cs
Comment (annelo-msft) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/IContainerRegistryAuthenticationClient.cs:12]: What motivates the change to this interface?  Why are we renaming the parameter names -- in ACR, it's important to distinguish between which accessToken is which because we use multiple: @Mohit-Chakraborty FYI

File: sdk/storage/Azure.Storage.Blobs/src/Generated/Models/BlobItemInternal.Serialization.cs
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/Generated/Models/BlobItemInternal.Serialization.cs:68]: @seanmcc-msft @amnguye do we have tests for this?

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/AvroSource.cs
Comment (annelo-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/AvroSource.cs:29]: Why are we changing an `IList<T>` parameter  to an `object`?

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/JsonFormat.cs
Comment (annelo-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/Models/JsonFormat.cs:31]: Why object here instead of a type?

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md
Comment (annelo-msft) [inline sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md:9]: Why did you change the package tag as part of this autorest change?

**Comment by annelo-msft** [comment]
It might be helpful to add a description to this PR, since it touches so many files, and people might have questions regarding what is changing and why that could be addressed there.

File: sdk/search/Azure.Search.Documents/src/autorest.md
Comment (AlexanderSher) [reply sdk/search/Azure.Search.Documents/src/autorest.md:19]: This flag affects method signatures, what has been a constant becomes a parameter. We probably can handle it differently, but we need to figure out how first.

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/api/Azure.Security.ConfidentialLedger.netstandard2.0.cs
Comment (AlexanderSher) [reply sdk/confidentialledger/Azure.Security.ConfidentialLedger/api/Azure.Security.ConfidentialLedger.netstandard2.0.cs:20]: We don't. This is an upgrade from the time when `Pageable` wasn't implemented.

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs
Comment (AlexanderSher) [reply sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs:61]: These tests have been doing manual iteration over pageable request, now it is handled by codegen. I'm not sure what can be tested here.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md
Comment (AlexanderSher) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md:9]: `package-artifacts-2019-06-01-preview` has an error that can't be mitigated by `transform` directive. So either we upgrade the tag or downgrade the version of used autorest package to the old one.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/IContainerRegistryAuthenticationClient.cs
Comment (AlexanderSher) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Authentication/IContainerRegistryAuthenticationClient.cs:12]: Another side effect of the named enums. Parameters aren't renamed - `PostContentSchemaGrantType grantType` is added as a first parameter, cause `PostContentSchemaGrantType` is now an enum.

File: sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md
Comment (annelo-msft) [reply sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/autorest.md:9]: Sounds good.  I have a change pending to upgrade to this tag, so I'll merge over this once it's committed, unless you'd like me to get that change in first.  Thanks!

File: sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/api/Azure.Monitor.Query.netstandard2.0.cs:135]: Do you need to set the flag for this library as well? I don't think we want these types.

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs
Comment (christothes) [reply sdk/confidentialledger/Azure.Security.ConfidentialLedger/tests/ConfidentialLedgerClientLiveTests.cs:61]: The tests validate that the API works as expected.

**Comment by AlexanderSher** [comment]
I think I'll create a different PR. It looks like introduced enums with single value have created too much disruption. I'll disable them by default.

**Comment by annelo-msft** [comment]
Would it make sense to hold this PR until after this release cycle?  If it's made at the beginning of a sprint, we'll have more time to handle any issues that come up before releasing.

File: sdk/storage/Azure.Storage.Blobs/src/Generated/Models/BlobItemInternal.Serialization.cs
Comment (kasobol-msft) [reply sdk/storage/Azure.Storage.Blobs/src/Generated/Models/BlobItemInternal.Serialization.cs:68]: https://github.com/Azure/azure-sdk-for-net/issues/24401

**Comment by AlexGhiondea** [comment]
@AlexanderSher tomorrow is the release day -- if there isn't anything needed for tomorrow, let's hold off on this update until after the packages are released.

**Comment by AlexanderSher** [comment]
Only if there is any package that uses LLC

**Pull request closed without merging**
